conda/common/url.py

```diff
@@ -403,7 +403,7 @@ def split_conda_url_easy_parts(known_subdirs, url):
     cleaned_url, token = split_anaconda_token(url)
     cleaned_url, platform = split_platform(known_subdirs, cleaned_url)
     _, ext = strip_pkg_extension(cleaned_url)
-    cleaned_url, package_filename = cleaned_url.rsplit('/', 1) if ext else (cleaned_url, None)
+    cleaned_url, package_filename = cleaned_url.rsplit('/', 1) if ext and '/' in cleaned_url else (cleaned_url, None)
```

Comment:
I am not at all convinced that this is the correct way to solve this crash.  We are allowing some very messed up URLs and then they fail once the hit the HTTP(S) machinery further along in execution.  I played around with creating an "InvalidUrl" exception but ran into a wall getting it to import correctly.

Side note, I think this import issue is why we're raising ValueError instead of a more meaningful error in this file but that's an adventure for another day.